---
layout: tutorial_hands_on
topic_name: genome_annotation
tutorial_name: strucutural_annotation_workflow
---

A genome is ready to be annotated within Apollo! To begin, the process of executing the structural annotation workflow will be detailed here.

> ### Agenda
>
> 1. Prerequisites
> 2. Workflow
> 3. Gene Calling
> 4. Completion
>
{: .agenda}

# Prerequisites 

<!-- BICH464 prerequisites discuss downloading the FASTA file from the Student Genomes Google Drive folder. -->

If it is a novel phage that has not been named yet, decide on a name for the phage. Access CPT Galaxy ([CPT Public Galaxy](https://cpt.tamu.edu/galaxy-pub), [CPT TAMU Galaxy](https://cpt.tamu.edu/galaxy)) to upload the FASTA file containing the desired phage genome. There are two methods to upload a FASTA file to Galaxy.
> 1. On the right side of the screen is a column titled “History;” it should be an empty, unnamed history containing only a blue box with a message. Click on the “… load your own data…” hyperlink.

![](../../images/structural-annotation-workflow-screenshots/(01)_New_History_Option.png)

> 2. On the left side of the screen is a column titled “Tools.” In the top right-hand corner of that column is an “upload” symbol. Click on that.

![](../../images/structural-annotation-workflow-screenshots/(0)_Upload_File.png)



# Part 1

Introduction about this part

## Subpart 1

Short introduction about this subpart.

<!--
{% icon hands_on %} will render the hands_on icon as specified in
_config.yml in the root of this repository.
-->

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon comment %} Comments
>    > A comment
>    {: .comment}
>
>    > ### {% icon tip %}Tip: A tip
>    >
>    > * Step1
>    > * Step2
>    {: .tip}
{: .hands_on}

## Subpart 2

Short introduction about this subpart.

> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. Step2
>
>    > ### {% icon question %} Question
>    >
>    > Question?
>    >
>    >    > ### {% icon solution %} Solution
>    >    >
>    >    > Answer to question
>    >    >
>    >    {: .solution}
>    >
>    {: .question}
{: .hands_on}

Some blabla
> ### {% icon hands_on %} Hands-on: Data upload
>
> 1. Step1
> 2. **My Tool** {% icon tool %} with the following parameters
>   - *"param1"*: the file `myfile`
>   - *"param2"*: `42`
>   - *"param3"*: `Yes`
>
> 3. **My Tool** {% icon tool %} with the following parameters
>   - {% icon param-text %} *"My text parameter"*: `my value`
>   - {% icon param-file %} *"My input file"*: `my file`
>   - {% icon param-files %} *"My multiple file input or collection"*: `my collection`
>   - {% icon param-select %} *"My select menu"*: `my choice`
>   - {% icon param-check %} *"My check box"*: `yes`
>
>    > ### {% icon question %} Questions
>    >
>    > 1. Question1?
>    > 2. Question2?
>    >
>    >    > ### {% icon solution %} Solution
>    >    >
>    >    > 1. Answer for question1
>    >    > 2. Answer for question2
>    >    >
>    >    {: .solution}
>    >
>    {: .question}
>
> 3. Step3
{: .hands_on}

# Part 2

Short introduction about this subpart.

> ### {% icon comment %} Comment
>
> Do you want to learn more about the principles behind mapping? Follow our [training](../../NGS-mapping)
{: .comment}


> ### {% icon details %} More details on the ....
>
> Add more details in Markdown. By default the box is collapsed. And is expanded when clicked
>
{: .details}

# Conclusion
{:.no_toc}

Conclusion about the technical key points. And then relation between the techniques and the biological question to end with a global view.
